bookshelf/src/controllers.php

```diff
@@ -139,3 +140,48 @@ function ($id) use ($app) {
         return new Response('', Response::HTTP_NOT_FOUND);
     }
 );
+
+$app->get('/login', function () use ($app) {
+    /** @var Google_Client $client */
+    $client = $app['google_client'];
+
+    $scopes = [ \Google_Service_Oauth2::USERINFO_PROFILE ];
+    $authUrl = $client->createAuthUrl($scopes);
+
+    return $app->redirect($authUrl);
+})->bind('login');
```

Comment:
the only one *required* to be named, though, is `login_callback`, because we have to generate that URL for oauth